### ISSUE TITLE
SelfLoader: status in core-p7 branch

### ISSUE THREAD

**Issue by jkeenan** [atoomic/perl#24]
In this ticket we will track our core-p7 progress on `dist/SelfLoader`.

As of the morning of Jul 08 2020, we are getting this output from `make test` for this distro.

```
t/../dist/SelfLoader/t/01SelfLoader ................................ Global symbol "$dir" requires explicit package name (did you forget to declare "my $dir"?) at t/01SelfLoader.t line 2.
Global symbol "$sep" requires explicit package name (did you forget to declare "my $sep"?) at t/01SelfLoader.t line 3.
Global symbol "$dir" requires explicit package name (did you forget to declare "my $dir"?) at t/01SelfLoader.t line 6.
Global symbol "$dir" requires explicit package name (did you forget to declare "my $dir"?) at t/01SelfLoader.t line 6.
Global symbol "$sep" requires explicit package name (did you forget to declare "my $sep"?) at t/01SelfLoader.t line 7.
Global symbol "$dir" requires explicit package name (did you forget to declare "my $dir"?) at t/01SelfLoader.t line 10.
Global symbol "$dir" requires explicit package name (did you forget to declare "my $dir"?) at t/01SelfLoader.t line 15.
Global symbol "$dir" requires explicit package name (did you forget to declare "my $dir"?) at t/01SelfLoader.t line 15.
Global symbol "$dir" requires explicit package name (did you forget to declare "my $dir"?) at t/01SelfLoader.t line 17.
Global symbol "$sep" requires explicit package name (did you forget to declare "my $sep"?) at t/01SelfLoader.t line 17.
Global symbol "$dir" requires explicit package name (did you forget to declare "my $dir"?) at t/01SelfLoader.t line 45.
Global symbol "$sep" requires explicit package name (did you forget to declare "my $sep"?) at t/01SelfLoader.t line 45.
BEGIN not safe after errors--compilation aborted at t/01SelfLoader.t line 71.
FAILED--no leader found
t/../dist/SelfLoader/t/02SelfLoader-buggy .......................... ok
t/../dist/SelfLoader/t/03taint ..................................... Bareword found where operator expected at t/01SelfLoader.t line 85, near "new Foo"
	(Do you need to predeclare new?)
Bareword found where operator expected at t/01SelfLoader.t line 105, near "new Foo"
	(Do you need to predeclare new?)
Bareword found where operator expected at t/01SelfLoader.t line 139, near "new Bar"
	(Do you need to predeclare new?)
Cannot run t/01SelfLoader.t: syntax error at t/01SelfLoader.t line 85, near "new Foo"
syntax error at t/01SelfLoader.t line 105, near "new Foo"
syntax error at t/01SelfLoader.t line 139, near "new Bar"
FAILED--expected 20 tests, saw 0


```
These results were generated on an unthreaded build on Linux.  The results might differ significantly if run on a threaded build.  There have been subsequent commits to this branch which may have impacted these results.  YMMV.

**Comment by atoomic** [comment]
this is fixed by 4cd6bcfa565b8b16998d304d42b4ae7de7e1b23f

**Comment by jkeenan** [comment]
Yes, as of v5.32.0-576-g9e5ecda8ae, SelfLoader is testing clean.  Thanks for your investigation.
